src/pudl_archiver/__init__.py

```diff
@@ -94,22 +80,24 @@ async def on_request_end(session, trace_config_ctx, params):
             cls = ARCHIVERS.get(dataset)
             if not cls:
                 raise RuntimeError(f"Dataset {dataset} not supported")
-            downloader = cls(session, only_years, download_directory=download_dir)
-            orchestrator = DepositionOrchestrator(
-                dataset,
-                downloader,
+            downloader = cls(
                 session,
-                upload_key,
-                publish_key,
-                dataset_settings_path=Path("dataset_doi.yaml"),
-                create_new=initialize,
-                dry_run=dry_run,
-                sandbox=sandbox,
-                auto_publish=auto_publish,
-                refresh_metadata=refresh_metadata,
+                run_settings.only_years,
+                download_directory=run_settings.download_dir,
+            )
+            depositor = await ZenodoPublishedDeposition.get_latest_version(
```

Comment:
That's a good point, the uninitialized state was kind of being hidden here. I still decided not to add a class for this state, given that it doesn't really have any meaningful "state", and all it does is go to draft state. I felt like adding another base class would just lead to clutter, so instead I added a level function called `get_deposition` that acts as an entry point into the state machine. This function either goes `published->draft` or creates a `draft` from scratch, but it will always return a draft that you can start working on.

Reply:
non-blocking, just musing: I guess the uninitialized state is actually sort of represented by the `DepositionInterface` wrapper class in this architecture - and then `get_deposition` actually represents the uninitialized -> draft transition.